code/modules/antagonists/voidwalker/voidwalker_traumas.dm

```diff
@@ -0,0 +1,129 @@
+/// Curse brain trauma that makes someone space textured, mute, pacifist and forbids them from entering space
+/datum/brain_trauma/voided
+	name = "Voided"
+	desc = "They've seen the secrets of the cosmis, in exchange for a curse that keeps them chained."
+	scan_desc = "cosmic neural pattern"
+	gain_text = ""
+	lose_text = ""
+	resilience = TRAUMA_RESILIENCE_LOBOTOMY
+	random_gain = FALSE
+	/// Type for the bodypart texture we add
+	var/bodypart_overlay_type = /datum/bodypart_overlay/texture/spacey
+	///traits we give on gain
+	var/list/traits_to_apply = list(TRAIT_MUTE, TRAIT_PACIFISM)
+	/// Do we ban the person from entering space?
+	var/ban_from_space = TRUE
+	/// What color we color the damage overlay?
+	var/damage_overlay_color_to_give = /obj/item/bodypart/head/voidwalker::damage_overlay_color
+
+/datum/brain_trauma/voided/on_gain()
+	. = ..()
+
+	owner.add_traits(traits_to_apply, TRAUMA_TRAIT)
+	if(ban_from_space)
+		owner.AddComponent(/datum/component/banned_from_space)
+	owner.AddComponent(/datum/component/planet_allergy)
+	RegisterSignal(owner, COMSIG_CARBON_ATTACH_LIMB, PROC_REF(texture_limb)) //also catch new limbs being attached
+	RegisterSignal(owner, COMSIG_CARBON_REMOVE_LIMB, PROC_REF(untexture_limb)) //and remove it from limbs if they go away
+
+	for(var/obj/item/bodypart as anything in owner.bodyparts)
+		texture_limb(owner, bodypart)
+
+	//your underwear is belong to us
+	if(ishuman(owner))
+		var/mob/living/carbon/human/human = owner //CARBON WILL NEVER BE REAL!!!!!
+		human.underwear = "Nude"
+		human.undershirt = "Nude"
+		human.socks = "Nude"
+
+	owner.update_body()
+
+/datum/brain_trauma/voided/on_lose()
+	. = ..()
+
+	owner.remove_traits(traits_to_apply, TRAUMA_TRAIT)
```

Comment:
Interestingly because this uses `TRAUMA_TRAIT` source it means that if someone has `/datum/brain_trauma/severe/mute` or `/datum/brain_trauma/severe/pacifism` this will effectively cure them of it